### PR TITLE
general instructions on how to create and style tables #208

### PR DESCRIPTION
**Pull request by janajaeger**
Add basic advice on tables.
- when to use which type of table
- basic code examples to build upon
- what styling options are available

### CODE REVIEW

File: xml/docu_styleguide.structure.xml [hunk 1]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+   </listitem>
+   <listitem>
+    <para>Complex layout constructs (screens, code) should not be used in tables.</para>
+   </listitem>
+  </itemizedlist>
+
+  <para>Some styling/style tips for tables:</para>
+  
+  <variablelist>
+   <varlistentry><term>Structure your table to have more rows than columns</term>
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2708]:
```suggestion
   <varlistentry><term>Structure the table to have more rows than columns</term>
```

File: xml/docu_styleguide.structure.xml [hunk 2]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+    <listitem>
+     <para>Humans prefer to skim for information that is aligned vertically. When designing 
+      tables, consider swapping rows and columns to allow for a better user experience.</para>
+    </listitem>
+   </varlistentry>
+   <varlistentry><term>Narrow down your columns</term>
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2714]:
```suggestion
   <varlistentry><term>Narrow down columns</term>
```

File: xml/docu_styleguide.structure.xml [hunk 3]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+    <listitem>
+     <para>Compress the data in your table as much as you possibly can. The less crowded 
+      the table cells are the better their content can be parsed by readers. Consider using icons,
+      add repeating words into the column title or using shorter number formats.</para>
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2718]:
```suggestion
     <para>Compress the data in your table as much as you possibly can. Table
      cells with less content are more easily parsed by readers. Consider using
      icons, adding repeating words into column titles, or using shorter number 
      formats.</para>
```

File: xml/docu_styleguide.structure.xml [hunk 4]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+    <para>Complex layout constructs (screens, code) should not be used in tables.</para>
+   </listitem>
+  </itemizedlist>
+
+  <para>Some styling/style tips for tables:</para>
+  
+  <variablelist>
+   <varlistentry><term>Structure your table to have more rows than columns</term>
+    <listitem>
+     <para>Humans prefer to skim for information that is aligned vertically. When designing
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2710]: 'Humans' stands a bit out here :-) Can we maybe use 'readers' like in the rest of the style guide?

File: xml/docu_styleguide.structure.xml [hunk 5]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+     <para>Compress the data in your table as much as you possibly can. The less crowded 
+      the table cells are the better their content can be parsed by readers. Consider using icons,
+      add repeating words into the column title or using shorter number formats.</para>
+    </listitem>
+   </varlistentry>
+   <varlistentry><term>Use colors to lead the reader's eyes</term>
+    <listitem>
+     <para>Use colors to make information stand out. If you just want to highlight small bits, 
+      use bright colors. If you need to color entire cells, use pastels. This option should only be
+      used with matrix type tables.</para>
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2725]:
```suggestion
     <para>Use colors to make the information stand out. If you just want to
      highlight small bits, use bright colors. If you need to color entire cells, 
      use pastels. This option should only be used with matrix type tables.</para>
```

File: xml/docu_styleguide.structure.xml [hunk 6]
```diff
@@ -2667,10 +2667,77 @@ xml:id="tab-install-source"</screen>
...
+     <!-- add list of supported colors as soon as toms' PR is merged -->
+    </listitem>
+   </varlistentry>
+   <varlistentry><term>Use striped rows</term>
+    <listitem>
+     <para>Color every second row in a light grey to make sure readers do not
```
Comment (dariavladykina) [inline xml/docu_styleguide.structure.xml:2731]:
```suggestion
     <para>Color every second row in light grey to make sure readers do not 
```

**Review by dariavladykina — COMMENTED** [review]
This reads very well, I just added some minor corrections.

File: xml/docu_styleguide.structure.xml (hunk 4 above)
Comment (janajaeger) [reply xml/docu_styleguide.structure.xml]: Done :)

File: xml/docu_styleguide.structure.xml [hunk 7]
```diff
@@ -2667,10 +2667,79 @@ xml:id="tab-install-source"</screen>
...
+    <para>Complex layout constructs (screens, code) should not be used in tables.</para>
+   </listitem>
+  </itemizedlist>
+
+  <para>Some style tips for tables:</para>
+  
+  <variablelist>
+   <varlistentry><term>Structure the table to have more rows than columns</term>
+    <listitem>
+     <para>Readers prefer to skim for information that is aligned vertically. When designing
```
Comment (tomschr) [inline xml/docu_styleguide.structure.xml:2710]: Maybe as an (optional) addition, you could mention:

```suggestion
     <para>Readers prefer to skim for information that is aligned vertically.
       It is easier to read a table without scrolling horizontally. When designing 
```

Or add it at the last sentence. Or don't change anything. Up to you.

File: xml/docu_styleguide.structure.xml [hunk 8]
```diff
@@ -2667,10 +2667,79 @@ xml:id="tab-install-source"</screen>
...
+   </listitem>
+   <listitem>
+    <para>Wordy explanations or descriptions should not be used in tables.</para>
+   </listitem>
+   <listitem>
+    <para>Complex layout constructs (screens, code) should not be used in tables.</para>
+   </listitem>
+  </itemizedlist>
+
+  <para>Some style tips for tables:</para>
```
Comment (tomschr) [inline xml/docu_styleguide.structure.xml:2705]: These tips are applicable to both "types", right? In that case, I would make it clear, they are general tips:

```suggestion
  <para>Some general style tips for tables:</para>
```

File: xml/docu_styleguide.structure.xml [hunk 9]
```diff
@@ -2667,10 +2667,79 @@ xml:id="tab-install-source"</screen>
...
+     <!-- add list of supported colors as soon as toms' PR is merged -->
+    </listitem>
+   </varlistentry>
+   <varlistentry><term>Use striped rows</term>
+    <listitem>
+     <para>Color every second row in light gray to make sure readers do not 
+      accidentally slip between lines. This should be the default for long 
+      tables. Do not use striped rows in matrix type tables. You may also consider disabling 
+      these stripes for short tables to not confuse readers.</para>
+     <!-- find out whether there is an easy way to do this instead of coloring each and every cell -->
```
Comment (tomschr) [inline xml/docu_styleguide.structure.xml:2737]: I'm not sure if we should mention that explicitly. This is usually done by the (CSS) stylesheet automatically.

**Review by tomschr — COMMENTED** [review]
Hi @janajaeger 
Great work! I have just some minor comments, nothing to worry about. It's up to you if you want to integrate it.
Thanks!

File: xml/docu_styleguide.structure.xml (hunk 7 above)
Comment (janajaeger) [reply xml/docu_styleguide.structure.xml:2710]: I prefer to leave stuff as it is :) But thanks for the suggestion anyway!

File: xml/docu_styleguide.structure.xml (hunk 9 above)
Comment (janajaeger) [reply xml/docu_styleguide.structure.xml:2737]: I'd still leave it in to provide some sort of rationale why we are using the stripes and when they don't add value.

**Review by jfaltenbacher — APPROVED** [review]
LGtM

**Review by PadraigDillon1 — APPROVED** [review]

**Pull request merged**